### PR TITLE
Sparse test sets

### PR DESCRIPTION
**Pull request by lbollar**
Referencing [#18801](https://github.com/JuliaLang/julia/issues/18801)

Added testsets to all of sparse.jl test file. For tests clearly mentioning issues, I have tried to group them in an issues testset with each subset testset containing an individual issue.

### CODE REVIEW

**Comment by tkelman** [comment]
Your merge commit undoes your change.

On line 127, `bitArray` should be spelled `BitArray`

**Comment by lbollar** [comment]
Apologies, still learning the pull request process. I'm guessing I needed to have rebased my branch instead of merge committing. What is best way to handle this now that I have submitted pull request already?

**Comment by tkelman** [comment]
I misread since github unhelpfully won't show a diff this big. It might have been better if split up into multiple commits of a few hundred lines each, but that's a consequence of bad tools. Viewing the diff as https://github.com/JuliaLang/julia/pull/18844/files?w=1 works better but @github won't let you leave review comments on that view.

The merge commit didn't hurt anything from what I can tell. You can address the review as additional commits to this branch in your fork, and we can squash upon merging.

**Comment by lbollar** [comment]
I have all the changes ready but for these two comments:

> line 325 is still testing ctranspose

> line 333 is still testing ftranspose!

Do these mean that titles of test sets on these lines should mention these functions?

**Comment by tkelman** [comment]
they do on master. you changed the titles so they don't any more

File: test/sparse/sparse.jl [hunk 1]
```diff
@@ -1164,22 +1167,22 @@ end
...
+        minj, maxj = extrema(J)
+        ((minj < 1) || (maxj > n)) && BoundsError()
     end
 
+    (alg == 0) ? Base.SparseArrays.getindex_I_sorted_bsearch_A(A, I, J) :
+    (alg == 1) ? Base.SparseArrays.getindex_I_sorted_bsearch_I(A, I, J) :
+    Base.SparseArrays.getindex_I_sorted_linear(A, I, J)
+end
+
+@testset "test_get_index_algs" begin
```
Comment (tkelman) [inline test/sparse/sparse.jl]: the function is called `test_getindex_algs`

**Comment by tkelman** [comment]
leave " for specialized matrix types" on line 1694

File: test/sparse/sparse.jl (hunk 1 above)
Comment (tkelman) [inline test/sparse/sparse.jl]: I meant you should remove the underscore from get_index, not drop the test_ prefix

**Comment by tkelman** [comment]
I pushed a minor revision to this branch. Note that lines 663-702 aren't related to issue #13024, and shouldn't be in the same testset.

**Comment by kshyatt** [comment]
@tkelman is this merge-able?

**Comment by tkelman** [comment]
not until the "shouldn't be in the same testset" comment is addressed

**Comment by lbollar** [comment]
Sorry, I misunderstood the meaning of your comment and thought you fixed this in your revision. I will take a look at this.

**Comment by lbollar** [comment]
I am guessing I mistakenly assumed all sections were commented due to the excellent commenting in file. I didn't see comments in original for these blocks, so I attempted to name them myself. These are probably awful testset names, so feel free to correct.

File: test/sparse/sparse.jl [hunk 2]
```diff
@@ -661,6 +661,53 @@ end
         end
     end
 
+    @testset "spones" begin
```
Comment (tkelman) [inline test/sparse/sparse.jl]: this still isn't an "issue" - by original location I meant where they were in the file before your PR, not at an earlier commit

**Comment by lbollar** [comment]
I moved them to after the first non issue related tests I could find in the original file where they took place. Sorry and thanks for patience.

**Comment by hayd** [comment]
bump! this needs a rebase.

**Comment by ViralBShah** [comment]
@lbollar Would be great if you could rebase to master and we should get this merged asap. Thanks for the work and patience.

**Comment by lbollar** [comment]
Traveling, but will try and get this done tonight when I get back home to
computer. Thanks for help.

On Nov 13, 2016 2:07 PM, "Viral B. Shah" <notifications@github.com> wrote:

@lbollar <https://github.com/lbollar> Would be great if you could rebase to
master and we should get this merged asap. Thanks for the work and patience.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/JuliaLang/julia/pull/18844#issuecomment-260208942>, or mute
the thread

**Comment by lbollar** [comment]
I am sorry all. I became very busy. The merge conflicts created over time are more difficult to solve than just using the latest commit and what I have here as a guideline to redo everything. Most of this pull request thread has been spent discussing naming and organization, which is mostly complete now, so the rewrite shouldn't that be painful. After discussing with @kshyatt, I think this is best route. Semester is ending so I should be able to rewrite soon, hopefully looking at doing it this weekend. Sorry for the delay.

**Comment by ViralBShah** [comment]
Thanks. Looking forward to having the new PR. Perhaps we should leave this one open in the meanwhile.

**Pull request closed without merging**